src/main/java/software/amazon/cloudwatchlogs/emf/logger/MetricsLogger.java

```diff
@@ -189,20 +192,56 @@ public MetricsLogger resetDimensions(boolean useDefault) {
      * @param key is the name of the metric
      * @param value is the value of the metric
      * @param unit is the unit of the metric value
+     * @param storageResolution is the resolution of the metric
      * @return the current logger
      * @throws InvalidMetricException if the metric is invalid
      */
-    public MetricsLogger putMetric(String key, double value, Unit unit)
+    public MetricsLogger putMetric(
+            String key, double value, Unit unit, StorageResolution storageResolution)
             throws InvalidMetricException {
         rwl.readLock().lock();
         try {
-            this.context.putMetric(key, value, unit);
+            this.context.putMetric(key, value, unit, storageResolution);
             return this;
         } finally {
             rwl.readLock().unlock();
         }
     }
 
+    /**
+     * Put a metric value. This value will be emitted to CloudWatch Metrics asynchronously and does
+     * not contribute to your account TPS limits. The value will also be available in your
+     * CloudWatch Logs
+     *
+     * @param key is the name of the metric
+     * @param value is the value of the metric
+     * @param storageResolution is the resolution of the metric
+     * @return the current logger
+     * @throws InvalidMetricException if the metric is invalid
+     */
+    public MetricsLogger putMetric(String key, double value, StorageResolution storageResolution)
```

Comment:
Not thread-safe

Reply:
This method just routes to [this method](https://github.com/awslabs/aws-embedded-metrics-java/pull/129/files/f4348977813388b799b068f5df0bd62c3c15163d#diff-1d7d3bb7bf7ba8bfaff971e4808b4b1c13431e66152b04bbd64ed0b6f860b163R199) which is thread safe. So I believe there won't be any problem. Correct me if I am wrong.